### PR TITLE
Provide initial parameter estimates using NCA via PKNCA

### PR DESCRIPTION
**Pull request by billdenney**
This is an initial version ready for discussion but not yet ready for merging.

### CODE REVIEW

**Comment by mattfidler** [comment]
Only comment I have so far is the writing a new estimation method should probably be in the `nlmr2est` package, I guess since that is where the estimation methods reside... What do you think?

**Comment by billdenney** [comment]
I'm okay either way.  This will add a couple of additional dependencies, and PKNCA "estimation" seems categorically different than other estimation methods since it's not minimizing an OFV.  So to me, it sits somewhere between nlmixr2est methods that are core math, external software, and an external support.

**Comment by billdenney** [comment]
Notes from the meeting:

* add vp, vp2, q, q2 multipliers as `pkncaControl()` options
* Allow custom parameter mapping (PKNCA ka = xxx) as a `pkncaControl()` option
* Allow `cp` variable detection as a `pkncaControl(dvParam="cp")` option

Add `nlmixr2.pkncaEst()` function for fitting the object and a print method for it:

```
nlmixr2.pkncaEst <- function(object, ...) {
  parse the object
  if (no data) {
  nlmixr2(object$ui, ...)
  } else {
  nlmixr2(object$ui, ...)
  }
}

print.pkncaEst <- function(x, ...) {
  cat("x$nca:\n")
  print(x$nca, ...)
  cat("x$ui\n")
  print(x$ui, ...)
}
```

File: R/convert.R [hunk 1]
```diff
@@ -333,16 +333,116 @@ bblDatToNonmem <- function(model, data, table=nlmixr2est::tableControl(), env=NU
...
+#' Determine standardized rxode2 column names from data
+#'
+#' @param data A data.frame as the source for column names
+#' @return A named character vector where the names are the standardized names
+#'   and the values are either the name of the column from the data or \code{NA}
+#'   if the column is not present in the data.
+#' @examples
+#' getStandardColNames(data.frame(ID=1, DV=2, Time=3, CmT=4))
+#' @export
+getStandardColNames <- function(data) {
```
Comment (billdenney) [inline R/convert.R:429]: @mattfidler, This is the function I'm thinking of for #52.  I'll give more description there.

**Comment by billdenney** [comment]
This is now very close to ready.  I need to add some tests, but it works in two examples for me.  It also has back-doors to allow the data for NCA to be different than the full model or to pass in PKNCA results that were precomputed (in case the automatic methods don't get it quite right).

It still needs a print method and its own `nlmixr2Est.pkncaEst` method.

**Comment by mattfidler** [comment]
Great news.  You may need to merge back to the main for the tests to work.

**Comment by mattfidler** [comment]
Only thing I saw is PKNCA is an import.

**Comment by billdenney** [comment]
I moved it to "suggests".  I think it's ready to merge (and to send to CRAN).

**Comment by billdenney** [comment]
Before sending it to CRAN, I have one more idea of how to make it more robust.  I'll commit that tonight and let you know when done.

**Comment by mattfidler** [comment]
Ok.  There are some issues I need to fix before cran anyhow.  I will merge this for now.  Create another pull request

**Pull request merged**